File: leet_2016_max_diff.js
/**
 * @param {number[]} nums
 * @return {number}
 */

var maximumDifference = function(nums) {
    let mini = Number.MAX_SAFE_INTEGER
    let res = -1
    for (let n of nums) {
        mini = Math.min(n, mini)
        if (n > mini) {
            res = Math.max(res, n - mini)
        }
    }
    return res
};

File: leet_2016_max_diff.py
class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        res = -1
        mini = 1e9 # keep record of the lowest so far
        for n in nums:
            mini = min(n, mini)
            if n > mini:
                res = max(res, n - mini)
        return res
